sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_providers/local_provider.py

```diff
@@ -63,6 +64,17 @@ def _raise_if_unsupported(self, operation, algorithm):
     def encrypt(self, algorithm, plaintext, iv=None):
         # type: (EncryptionAlgorithm, bytes, Optional[bytes]) -> EncryptResult
         self._raise_if_unsupported(KeyOperation.encrypt, algorithm)
+
+        # If an IV isn't provided with AES-CBCPAD encryption, try to create one
+        if iv is None and algorithm.value.endswith("CBCPAD"):
+            try:
+                iv = os.urandom(16)
+            except NotImplementedError as ex:
+                raise RuntimeError(
```

Comment:
Should this be a `ValueError` i.e., we require the value of `iv` not be `None` on this platform?

Reply:
That's a great idea -- I was struggling to think of which exception type would work best, but that makes sense.